src/System.IO.Pipelines/src/System/IO/Pipelines/Pipe.cs

```diff
@@ -206,12 +206,19 @@ private BufferSegment AllocateSegment(int sizeHint)
 
             if (_pool is null)
             {
+                // Use the array pool
                 newSegment.SetMemory(ArrayPool<byte>.Shared.Rent(GetSegmentSize(sizeHint)));
             }
-            else
+            else if (sizeHint <= _pool.MaxBufferSize)
             {
+                // Use the specified pool if it fits
                 newSegment.SetMemory(_pool.Rent(GetSegmentSize(sizeHint, _pool.MaxBufferSize)));
             }
+            else
+            {
+                // We can't use the pool so allocate an array
+                newSegment.SetUnownedMemory(new byte[sizeHint]);
```

Comment:
Do we wan't to add `StrictMode` to pipe options that would check that pool is always used?

Reply:
Nope, not until we think we need it. I was thinking about that but I think it's a bit early to do that.